engine/StoryState.js

```diff
@@ -312,31 +299,45 @@ export class StoryState{
 			}
 		});
 	}
-	
-	MatchRightGlueForLeftGlue(leftGlue){
-		if (!leftGlue.isLeft) return null;
-		
-		for (var i = this._outputStream.length - 1; i >= 0; i--) {
-			var c = this._outputStream[i];
-//			var g = c as Glue;
-			var g = c;
-			if (g instanceof Glue && g.isRight && g.parent == leftGlue.parent) {
-				return g;
-			} else if (c instanceof ControlCommand) // e.g. BeginString
-				break;
+
+	CleanOutputWhitespace (str){
+		var sb = new StringBuilder(str.Length);
+
+		var currentWhitespaceStart = -1;
+
+		for (var i = 0; i < str.Length; i++) {
+			var c = st.charAt(i);
+
+			var isInlineWhitespace = (c == ' ') || (c == '\t');
+
+			if (isInlineWhitespace && currentWhitespaceStart == -1)
+				currentWhitespaceStart = i;
+
+			if (!isInlineWhitespace) {
+				if (c != '\n' && currentWhitespaceStart > 0) {
+					sb.Append(str.substr(currentWhitespaceStart, i - currentWhitespaceStart));
```

Comment:
This certainly needs a more thorough check, `substr` should do exactly the same thing as `Substring`, but we can't be too careful.